final_src/final_src/5_entities/post/post.api.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  BaseFilterParams,
+  DeletedPost,
+  GetPostsResponse,
+  ParamsWithSearch,
+  ParamsWithTag,
+  Post,
+  POST_CONSTANTS,
+  SORT_ORDER,
+} from '@/entities/post';
+import { ApiService } from '@/shared/lib';
+
+export const getPosts = async ({
+  limit = POST_CONSTANTS.DEFAULT_LIMIT,
+  skip = POST_CONSTANTS.DEFAULT_SKIP,
+  sortBy,
+  sortOrder,
+}: BaseFilterParams): Promise<GetPostsResponse> => {
+  const params = {
+    limit: limit.toString(),
+    skip: skip.toString(),
+    sortBy: sortBy || '',
+    sortOrder: sortOrder || SORT_ORDER.ASC,
+  };
+
+  return ApiService.get<GetPostsResponse>('/posts', params);
+};
+
+export const getPostsByTag = async ({
+  selectedTag,
+  limit,
+  skip,
+  sortBy,
+  sortOrder,
+}: ParamsWithTag): Promise<GetPostsResponse> => {
+  const params = {
+    limit: limit?.toString() || '0',
+    skip: skip?.toString() || '0',
+    sortBy: sortBy || '',
+    sortOrder: sortOrder || SORT_ORDER.ASC,
+  };
+
+  return ApiService.get<GetPostsResponse>(`/posts/tag/${selectedTag}`, params);
+};
+
+export const getPostsBySearch = async ({
+  searchQuery,
+  limit,
+  skip,
+  sortBy,
+  sortOrder,
+}: ParamsWithSearch): Promise<GetPostsResponse> => {
+  const params = {
+    limit: limit?.toString() || '0',
+    skip: skip?.toString() || '0',
+    searchQuery,
+    sortBy: sortBy || '',
+    sortOrder: sortOrder || SORT_ORDER.ASC,
+  };
+
+  return ApiService.get<GetPostsResponse>('/posts/search', params);
+};
+
+export const createPost = async (newPost: Omit<Post, 'id'>): Promise<Post> => {
+  return ApiService.post<Post>('/posts/add', newPost);
+};
```

Comment:
공통된 내용인데, ApiService.post 에 이미 제네릭으로 Post 가 반환 타입임을 적용했으므로 래핑한 함수에서는 굳이 명시하지 않아도 알 수 있을 것 같아서 간단하게 이렇게만 적어도 괜찮을 것 같아요!
```suggestion
export const createPost = async (newPost: Omit<Post, 'id'>) => {
  return ApiService.post<Post>('/posts/add', newPost);
};
```